src/curl.c

```diff
@@ -437,7 +442,19 @@ static int cc_config_add_page(oconfig_item_t *ci) /* {{{ */
       status = cf_util_get_string(child, &page->plugin_name);
     else if (strcasecmp("URL", child->key) == 0)
       status = cf_util_get_string(child, &page->url);
-    else if (strcasecmp("User", child->key) == 0)
+    else if (strcasecmp("AddressFamily", child->key) == 0) {
+      status = cf_util_get_string(child, &af);
+      if (status != 0 || af == NULL) {
+        page->address_family = CURL_IPRESOLVE_WHATEVER;
```

Comment:
Please don't set `address_family` on failure. This is impossible to debug for users: for some reason, they run into an internal error and instead of an error message name resolution works slightly different – that's a nightmare to track down.

Instead, just `continue` – the `cf_util_get_…` functions all report an error when returning non-zero.

Reply:
I made this into a `WARNING()` and made all warning messages more verbose (including plugin instance name, for those who configure more than one `<Page>` block).
I also added explicit handling of `any` and unknown values for `AddressFamily`